niceman/retrace/packagemanagers.py

```diff
@@ -179,6 +271,40 @@ def _create_package(self, pkgname):
         lgr.debug("Found package %s", pkg)
         return pkg
 
+    class MultipleReleaseFileMatch(RuntimeError):
+        pass
+
+    def _find_release_date(self, site, archive):
+        if not site:
+            return None
+        # First find the release file
+        # This uses logic similar to apt.utils.get_release_filename_for_pkg
+        # but restricts matches to the source site
+        rfile = None
+        dirname = apt_pkg.config.find_dir("Dir::State::lists")
+        # If we want to avoid using cache._list, we can call
+        # apt_pkg.SourceList() directly
+        for uri in set([metaindex.uri for metaindex in cache._list.list]):
+            if site == urlparse(uri).netloc:
```

Comment:
after we have this merged might be worth checking if performs correctly in case of local repos: https://github.com/ReproNim/niceman/issues/59